Gunless cube holding

LarsMans64 opened the pull request:
What i did:
- You can now hold cubes without a gun
- Fixed all known issues where the gun desynchronizes with the cube
  - The gun now checks for this on its own instead of the spaghetti code it used to be
  - Stuff is now handled by the entity itself instead of by the gun item
- Tweaked particles

To do:

- [x] Tweak particles
- [x] Handle switching items (hand swap, drop item, scroll, etc.)
- [x] Fix holding sound playing indefinitely if picking up a cube with 2 guns simultaneously
- [x] Support for lefties

Perhaps lock the cube on the angle it was picked up at, instead of snapping to the looking direction

Resolves #64 
Resolves #52

LarsMans64 committed:
Gunless cube holding

LarsMans64 committed:
yeet

phantomeye1931 commented:
I think the cube rotation shouldn't be locked. I'd say it's fine as it is, it may be buggy but I haven't really noticed that actually being an issue except for turrets acting a little janky sometimes
Note to self: Maybe that can be fixed by modulating the wing interpolation?

LarsMans64 committed:
portal gun figures out whether it is holding something on its own now

LarsMans64 committed:
portal gun sparks tweaked

LarsMans64 committed:
comment

LarsMans64 committed:
i quit

LarsMans64 committed:
ok i didnt completely fix it but at least it cannot go on forever anymore

phantomeye1931 commented:
seems fine

LarsMans64 commented:
I'll rebase this in a minute

LarsMans64 committed:
Merge branch 'master' into cube-holding

LarsMans64 committed:
fix merge issue

phantomeye1931 reviewed (commented):
I'm going through this in-game, and so far this seems really good. But this would also be a great moment to limit the grab range. This is some unholy reach that'd definitely going to ruin puzzles

<img width="1472" height="854" alt="image" src="https://github.com/user-attachments/assets/37fdf6a3-991d-4388-8b04-cedf895eb3ae" />

LarsMans64 commented:
could you do that

LarsMans64 commented:
the reach for pressing buttons was lowered already, you could make this the same perhaps

phantomeye1931 commented:
Also, cubes can be picked up and dropped in spectator mode
<img width="844" height="475" alt="image" src="https://github.com/user-attachments/assets/f45980e8-9cf8-4a71-bd88-04d8b9dd19e7" />

phantomeye1931 commented:
> could you do that

sure

phantomeye1931 committed:
Unified reach for picking up cubes and pressing buttons
Yes the constant is in PortalGun.java, suck my balls

phantomeye1931 committed:
Prevent cubes from being interacted with while in spectator
You can still hold onto cubes while transitioning into spectator, but I cannot be bothered.

phantomeye1931 reviewed (approved):
I solved my issues with this branch anything left? Or is this ready to be merged @LarsMans64

LarsMans64 commented:
yeah lets merge, i need this on main so bad
